contracts/examples/bonding-curve/src/curves/curve_function.rs

```diff
@@ -0,0 +1,33 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use crate::function_selector::*;
+pub trait CurveFunction<BigUint: BigUintApi>
+where
+	for<'a, 'b> &'a BigUint: core::ops::Add<&'b BigUint, Output = BigUint>,
+	for<'a, 'b> &'a BigUint: core::ops::Sub<&'b BigUint, Output = BigUint>,
+	for<'a, 'b> &'a BigUint: core::ops::Mul<&'b BigUint, Output = BigUint>,
+	for<'a, 'b> &'a BigUint: core::ops::Div<&'b BigUint, Output = BigUint>,
+	for<'b> BigUint: core::ops::AddAssign<&'b BigUint>,
+	for<'b> BigUint: core::ops::SubAssign<&'b BigUint>,
+	for<'b> BigUint: core::ops::MulAssign<&'b BigUint>,
+	for<'b> BigUint: core::ops::DivAssign<&'b BigUint>,
+	BigUint: BigUintApi,
+{
+	fn function(
+		&self,
+		token_start: BigUint,
+		amount: BigUint,
+		arguments: &CurveArguments<BigUint>,
+	) -> SCResult<BigUint>;
+
+	fn sell(&self, amount: BigUint, arguments: CurveArguments<BigUint>) -> SCResult<BigUint> {
```

Comment:
this file exposes primitive BigUint ops for a curve. consider renaming the `buy` and `sell` funcs or extract them to a contract concerned with token processing